src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -573,6 +576,9 @@ def build_vmss_resource(name, naming_prefix, location, tags, overprovision, upgr
     if custom_data:
         os_profile['customData'] = b64encode(custom_data)
 
+    if secrets:
+        os_profile['secrets'] = secrets
```

Comment:
One could, but we'd have to introduce `--certificate-store` on the VM command. I'm not entirely against this, but it would limit the certificate stores the secrets are added to.

Beyond just certificate stores, if another property is added to the vaultCertificates array objects such as `fileName` or something else, then it would cause us to need to add something like `--file-names [array of file names]` to the VM / VMSS create commands. I think it's less scary to extend `vm-format` to produce:

```json
[{ 
  "sourceVault": { 
    "id": "value" 
  }, 
  "vaultCertificates": [{ 
    "certificateUrl": "value", 
    "certificateStore": "cert store name (only on windows)",
    "fileName": "somecert"
  }]
}]
```